src/expint.jl

```diff
@@ -208,9 +208,10 @@ En_safe_gamma_term(ν::Integer, z::Real) = (z ≥ 0 || isodd(ν) ? 1 : -1) * exp
 # returns the two terms from the above equation separately
 function En_cf_gamma(ν::Number, z::Number, n::Int=1000)
     A = float(1 - ν)
-    B::typeof(A) = 1
-    Bprev::typeof(A) = 0
-    Aprev::typeof(A) = 1
+    A, _ = promote(A, z)
```

Comment:
It might be better to first promote and then turn `A` (and `z`?) to floating point numbers: If `ν isa Int` and `z isa Float32`, the implementation in the PR would yield `A isa Float64`, whereas first promoting and then converting to floating point numbers would give you `A isa Float32`. For instance, you could use something like
```suggestion
    A, z = map(float, promote(1 - ν, z))
```